lib/src/io_utils.dart

```diff
@@ -118,40 +120,102 @@ final RegExp partOfRegexp = RegExp('part of ');
     'as Dartdoc 1.0.0')
 final RegExp newLinePartOfRegexp = RegExp('\npart of ');
 
-/// Best used with Future<void>.
-class MultiFutureTracker<T> {
-  /// Approximate maximum number of simultaneous active Futures.
-  final int parallel;
+typedef TaskQueueClosure<T> = Future<T> Function();
+
+class _TaskQueueItem<T> {
+  _TaskQueueItem(this._closure, this._completer, {this.onComplete});
+
+  final TaskQueueClosure<T> _closure;
+  final Completer<T> _completer;
+  void Function() onComplete;
+
+  Future<void> run() async {
+    try {
+      final result = await _closure();
+      if (result != null) {
```

Comment:
Hmm... can we skip the if/else and just `_completer.complete(result);`?

Reply:
Duh.  Yes, removed.